CEGameApp.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using CEGameApp.API.Data;
+using CEGameApp.API.Dtos;
+using CEGameApp.API.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace CEGameApp.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthRepository _repo;
+        private readonly IConfiguration _config;
+
+        public AuthController(IAuthRepository repo, IConfiguration config)
+        {
+            _repo = repo;
+            _config = config;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserForRegisterDto userForRegisterDto)
+        {
+            //validate request
+            userForRegisterDto.UserName = userForRegisterDto.UserName.ToLower();
```

Comment:
this is a bit of a code smell.  you are Mutating this object inline.  ie, you are converting this UserName ToLower, completely changing the value of this Property, consider just passing in a LowerCase version of this string directly into the UserExists check

Reply:
Fixed.